src/js/heuristicblocking.js

```diff
@@ -157,6 +158,31 @@ HeuristicBlocker.prototype = {
     );
   },
 
+  /**
+   * Record third-party pings as tracking.
+   */
+  pingAccounting: function (details) {
+    const fqdn = (new URI(details.url)).host,
+      origin = window.getBaseDomain(fqdn),
+      tab_origin = tabOrigins[details.tabId];
```

Comment:
This is probably susceptible to navigation-related misattribution (`tabOrigins` has the new site's URL instead of actual URL, similar issue to #1997), but this may be OK for now (problem exists elsewhere, no need to block new feature).

Reply:
I think this is an issue for the `ping` attribute on hyperlinks. For example, visit google.com in chrome, make a search, and click on any outbound link. For me, most of the time clicking on `foo.com` adds that domain as an entry for `google.com` in the `snitch_map`. Since the ping request fires off simultaneously with the user navigating away from Google, this bug is much more likely to come up.

Reply:
True, but is that a blocking issue, practically speaking? Privacy Badger is very likely to learn to [block `google.com`](https://github.com/EFForg/badger-sett/commit/302de8f35446f8f3f406b5e4bc5643ea70da337b#diff-15d7fd2929994a7ad02003208db9ecf4R227) anyway.

Reply:
I guess in general the navigation bug means we are likely to record first-party pings as third-party tracking. Or third-party pings as third-party on the wrong party.

Reply:
Yeah, this navigation-related attribution bug doesn't affect the tracking domain, just where it was seen. We mis-attribute tracking to the wrong site. #1997 is waiting on information from Chromium devs at the moment.